src/NumFieldOrd/NfOrd/NfOrd.jl

```diff
@@ -1046,18 +1048,50 @@ function _order(K::S, elt::Vector{T}; cached::Bool = true, check::Bool = true, e
     end
 
     start = 1
+    old_length = length(bas)
+    # Commutative case:
     # We only multiply the elements of index start:length(bas) by e .
     # Example: bas = [a_1, ..., a_k] with a_1 = 1. Then
     # new_bas := [e, e*a_2, ..., e*a_k] and we append this to bas and set
     # start := k + 1. In the next iteration, we then have
     # new_bas := [e^2, e^2*a_2, ..., e^2*a_k] (assuming that there was no
     # reduction of the basis in between).
-    for i in 1:df
+
+    powers_of_e = 1 # count the iterations, so the power of e by which we multiply
+                    # (only relevant in the commutative case)
+    while true
+      # In the commutative case, this behaves like a "for _ in 1:df"-loop
+      if is_commutative(K) && powers_of_e == df + 1
```

Comment:
Maybe move the check to the start of the for loop? I know it should be cached, but still.

Reply:
I'm not sure I understand. Do you mean putting the result of `is_commutative(K)` in a local variable?